src/resdk/tables/variant.py

```diff
@@ -0,0 +1,424 @@
+
+        self.discard_fakes = discard_fakes
+
+        self.mutations = mutations
+        if mutations is not None:
+            if not isinstance(mutations, list):
+                raise ValueError("Mutations must be a list of strings")
+        else:
+            # Check for heterogeneous mutations only if mutations is not specified
+            self.check_heterogeneous_mutations()
+
+    def check_heterogeneous_mutations(self):
+        """Ensure variants are selected for the same mutations in all samples."""
+        mutations = set([str(d.input["mutations"]) for d in self._data])
+
+        if len(mutations) > 1:
+            raise ValueError(
+                "Variants should be computed with the same mutation input. "
+                f"Variants of samples in collection {self.collection.name} "
+                f"have been computed with {', '.join(list(mutations))}.\n"
+                "Use mutations filter in the VariantTables constructor.\n"
+            )
+
+    def parse_mutations_string(self, mutations):
+        """Parse mutations string."""
+        db = {}
+        for mutation in mutations:
+            if ":" not in mutation:
+                # Only gene is given
+                gene = mutation
+                aa_changes = set()
+            else:
+                gene, aa_changes = mutation.split(":")
+                aa_changes = set(map(str.strip, aa_changes.split(",")))
+
+            db[gene] = aa_changes
+
+        return db
+
+    def is_mutation_subset(self, first, second):
+        """Check if mutations given in first are a a subset of second."""
+        db_first = self.parse_mutations_string(first)
+        db_second = self.parse_mutations_string(second)
+
+        for first_gene, first_aas in db_first.items():
+            if first_gene not in db_second:
+                return False
+            if not db_second[first_gene]:
+                # If the second covers the whole gene, all is good
+                continue
+            second_aas = db_second[first_gene]
+            if not first_aas and second_aas:
+                # First covers the whole gene, second does not
+                return False
+            if not first_aas.issubset(second_aas):
+                return False
+
+        return True
+
+    @property
+    @lru_cache()
+    def _data(self) -> List[Data]:
+        """Fetch data objects.
+
+        Fetch Data of type ``self.process_type`` from given collection
+        and cache the results in memory. Only the needed subset of
+        fields is fetched.
+
+        :return: list of Data objects
+        """
+        data = []
+        sample_ids, repeated_sample_ids = set(), set()
+        for datum in self.collection.data.filter(
+            type=self.process_type,
+            status="OK",
+            ordering="-created",
+            fields=self.DATA_FIELDS,
+        ):
+            # 1 Filter by mutations, if required
+            if self.mutations:
```

Comment:
Currently, if you call `VariantTables` and set a certain "gene: aminoacid" combination that is not present in the target samples as an input to `mutations`, you will get an error `Collection <collection> has no data:mutationstable data!`, even though data objects with type `data:mutationstable` exist in this collection. It would be better if this error messages were more informative. Example of an error: 

```
vt = VariantTables(c, mutations=["TP53: Ser"])
vt.variants
ValueError: Collection <collection> has no data:mutationstable data!
```